packages/ckeditor5-image/src/image/imageblocktoinlinecommand.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module image/image/imageinlinetoblockcommand
+ */
+
+import Command from '@ckeditor/ckeditor5-core/src/command';
+import { findOptimalInsertionPosition } from '@ckeditor/ckeditor5-widget/src/utils';
+import { isImage } from './utils';
+
+/**
+ * The image inline command. It is used to convert block image io inline image.
+ *
+ * @extends module:core/command~Command
+ */
+export default class ImageBlockToInlineCommand extends Command {
+	/**
+	 * @inheritDoc
+	 */
+	refresh() {
+		const element = this.editor.model.document.selection.getSelectedElement();
+
+		this.isEnabled = isImage( element );
```

Comment:
Maybe this command should be disabled also if the schema does not allow paragraphs with inline images in the selection position?

Reply:
But if in this position there is a block image, then always also paragraph with inline image will be allowed - or there is any exception?

Reply:
In the current schema configuration that's true, but some integration/feature might disable some cases.

Reply:
I was wrong in the previous message. In the selected position the converted image will never be allowed, but the conversion command modifies the DOM to allow it. In the block to inline conversion, it adds `paragraph` to allow inserting `inlineImage`. So proposed checking makes no sense.